single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/LabelProvider.java

```diff
@@ -90,6 +93,8 @@ enum LabelDirection {
 
     List<ElectricalNodeInfo> getElectricalNodesInfos(VoltageLevelGraph graph);
 
+    List<NodeLegend> getElectricalNodeInfoNodes(ElectricalNodeInfo nodeInfo);
```

Comment:
We should merge these two methods together. An `electricalNodeInfo` would contain a list of `captions`, to avoid this double call to `LabelProvider`. 

You could transform the `ElectricalNodeInfo` class into a record:
```java
public record ElectricalNodeInfo(String busId, List<Caption> captions) {
}
```

And maybe, as we're breaking things here, take the opportunity to rename it to `BusInfo`, as `ElectricalNode` is not a word used in powsybl world.

Reply:
So at the end `ElectricalNodeInfo` could be like this:

```java
public record BusInfo(String busId, List<Caption> captions) {
    public record Caption(String label, String type) {
    }
}
```

Reply:
done, BusInfo is already existing in the same package
I've name it NodeInfo waiting for a better solution

Reply:
"type" is only used for creating the svg id, by not having an enum it lets the user add any kind of information.

About the naming, I forgot `BusInfo` was already taken!
Instead of `NodeInfo`, shouldn't we call it `BusLegendInfo`? `LegendBusInfo`? `BusCaptions`?

Reply:
fixed, renamed by BusLegendInfo